makefile

```diff
@@ -1,20 +1,132 @@
-# -----------------------------------------------------------------------------
-# Copyright &copy; 2015 Ben Blazak <bblazak@fullerton.edu>
-# Released under the [MIT License] (http://opensource.org/licenses/MIT)
-# -----------------------------------------------------------------------------
+/*# -----------------------------------------------------------------------------
+# Copyright &copy; 2015 Paul Sihavong <paul.sihavong@csu.fullerton.edu>
+# Released under the[MIT License](http://opensource.org/licenses/MIT)
+# -----------------------------------------------------------------------------*/
 
-EXE := $(wildcard *.cpp)
-EXE := $(EXE:%.cpp=%)
+/*Program Description
+*The purpose of this program is to demonstrate the use of recursive and iterative
+functions. Also examining the code, there is a difference in the method of
+iterative and recursive functions.*/
 
-CXXFLAGS := -std=c++11
-# CXXFLAGS += -pedantic
-# CXXFLAGS += -Weverything
+#include <iostream>
+#include <cmath>
+#include <string>
+using std::cout;
+using std::cin;
+using std::endl;
+using std::string;
 
-# -----------------------------------------------------------------------------
-.PHONY: all clean
+int gcd_iterative(int a, int b){
+	a = abs(a);
+	b = abs(b);
+	if (a > b)
+		a -= b;
+	else if (a < b)
+		b -= a;
+	else if (a == 0)
+		return b;
+	else if (b == 0)
+		return a;
+}
 
-all: $(EXE)
+int gcd(int a, int b) {
+	a = abs(a);
+	b = abs(b);
+	if (a == 0)
+		return b;
+	else if (b == 0)
+		return a;
+	else if (a > b)
+		return gcd(a - b, b);
+	else if (a < b)
+		return gcd(a, b - a);
+}
 
-clean:
-	rm $(EXE)
+string int_to_RN(int n){ // This recursion will be referring to its iterative
+	if (n > 1000){		 // form from Assignment 1
+		int_to_RN(n - 1000);
+		return "M";
```

Comment:
should be `return "M" + int_to_RN(n-1000);` -- as written, the return value of the recursive call is discarded, and only a single "M" will be returned.